polars/polars-arrow/src/kernels/rolling/nulls.rs

```diff
@@ -58,6 +60,67 @@ where
     ))
 }
 
+#[allow(clippy::too_many_arguments)]
+fn rolling_apply_quantile<T, Fo, Fa>(
+    values: &[T],
+    bitmap: &Bitmap,
+    quantile: f64,
+    interpolation: QuantileInterpolOptions,
+    window_size: usize,
+    min_periods: usize,
+    det_offsets_fn: Fo,
+    aggregator: Fa,
+) -> ArrayRef
+where
+    Fo: Fn(Idx, WindowSize, Len) -> (Start, End) + Copy,
+    // &[T] -> values of array
+    // &[u8] -> validity bytes
+    // QuantileInterpolOptions -> Interpolation option
+    // usize -> offset in validity bytes array
+    // usize -> min_periods
+    Fa: Fn(&[T], &[u8], f64, QuantileInterpolOptions, usize, usize) -> Option<f64>,
+{
+    let len = values.len();
+    let (validity_bytes, offset, _) = bitmap.as_slice();
+
+    let mut validity = match create_validity(min_periods, len as usize, window_size, det_offsets_fn)
+    {
+        Some(v) => v,
+        None => {
+            let mut validity = MutableBitmap::with_capacity(len);
+            validity.extend_constant(len, true);
+            validity
+        }
+    };
+
+    let out = (0..len)
+        .map(|idx| {
+            let (start, end) = det_offsets_fn(idx, window_size, len);
+            let vals = unsafe { values.get_unchecked(start..end) };
+            match aggregator(
+                vals,
+                validity_bytes,
+                quantile,
+                interpolation,
+                offset + start,
+                min_periods,
+            ) {
+                Some(val) => val,
+                None => {
+                    validity.set(idx, false);
+                    f64::default()
+                }
+            }
+        })
+        .collect_trusted::<Vec<f64>>();
+
+    Arc::new(PrimitiveArray::from_data(
```

Comment:
I think we should make this generic over the `Float` trait. We expect rolling functions over `f32` to stay of that dtype.

Reply:
This should be resolved, I didn't use the Float trait per se but the outcome is the same